### DISCUSSION TITLE
Testing a proxy via VPN client (and, probably, via QEMU)

### DISCUSSION THREAD

**Discussion by SuperMaxusa** [benjamincburns/websockproxy#27] [Q&A] [answered]
Hello, I saw in README that websockproxy can be connected via VPN client. Is there an instruction how to do this?
https://github.com/benjamincburns/websockproxy/blob/27ce0d2db9d9f332e3745dec7428ff4786bd8b0c/README.md?plain=1#L34-L35

And some question about QEMU, I have some ideas about it:
1. Connect QEMU to same tap device as websockproxy on the host: https://gist.github.com/extremecoders-re/e8fd8a67a515fee0c873dcafc81d811c#example-tap-network, https://wiki.archlinux.org/title/QEMU#Tap_networking_with_QEMU
2. QEMU can open [TCP/UDP socket](https://john-millikin.com/improved-unix-socket-networking-in-qemu-7.2) and send/receive Ethernet frames over this socket, but then it needs a bridge `QEMU socket <--> WebSocket`.

Which of these is best for debugging/testing?

Sorry if the questions are incorrect and thanks for any reply!

**Comment by benjamincburns** [comment, marked as answer]
I probably shouldn't have included that note in the README, sorry. There _are_ websocket VPN clients out there that likely can connect to websockproxy just fine, but I haven't tested any of them myself. They'd just need to work by transmitting and receiving ethernet frames via the websocket in the same way that jor1k does with websockproxy.
